units/src/parse.rs

```diff
@@ -19,14 +20,18 @@ use crate::prelude::*;
 #[derive(Debug, Clone, PartialEq, Eq)]
 #[non_exhaustive]
 pub struct ParseIntError {
-    pub(crate) input: String,
-    // for displaying - see Display impl with nice error message below
+    /// The input string that caused the error.
+    pub input: String,
```

Comment:
I'd rather pull all other types together with `ParseIntError` in a single commit than expose it.

Reply:
I made the fields private and added a getter for the source error (we had one already for the input string). Adds a clone in `rust-bitcoin` but its a step in the right direction at least.